ethereum/contracts/state-transition/chain-deps/facets/Mailbox.sol

```diff
@@ -404,7 +513,14 @@ contract MailboxFacet is Base, IMailbox {
         require(_message.length >= 56, "pm");
 
         (uint32 functionSignature, uint256 offset) = UnsafeBytes.readUint32(_message, 0);
-        require(bytes4(functionSignature) == this.finalizeEthWithdrawal.selector, "is");
+        require(
+            bytes4(functionSignature) ==
+                bytes4(
+                    // Note this is the selector with the chainId, as that is called from bridgehub
+                    abi.encodeWithSignature("finalizeEthWithdrawal(uint256,uint256,uint256,uint16,bytes,bytes32[])")
```

Comment:
This message actually comes from the L2Eth system contract: https://github.com/matter-labs/era-system-contracts/blob/d42f707cbe6938a76fa29f4bf76203af1e13f51f/contracts/L2EthToken.sol#L123, so you should provide the selector without the chainId

Reply:
If we move the withdrawals to the Bridgehub, I think it should stay as it is.

Reply:
True, corrected